elide-core/src/main/java/com/yahoo/elide/core/dictionary/EntityDictionary.java

```diff
@@ -2001,9 +2002,16 @@ public static String getEntityName(Type<?> modelClass) {
     }
 
     public static <T> Type<T> getType(T object) {
-        return object instanceof Dynamic
-                ? ((Dynamic) object).getType()
-                : new ClassType(object.getClass());
+        if (object instanceof Dynamic) {
+            return ((Dynamic) object).getType();
+        } else {
+            ClassType<T> classType = (ClassType<T>) TYPE_MAP.computeIfAbsent(
+                    object.getClass(),
+                    x -> new ClassType<>(object.getClass())
```

Comment:
Sorry that lambda was not very smart from me because I was checking with the cast first. This should be ClassType::new instead (since `x == object.getClass()` anyway)